File: src/problems/014. Longest Common Prefix/Solution.java
// 14. Longest Common Prefix
// Accepted 10ms

class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 0)
            return "";
        
        
        int dig = 0;
        boolean continueFlag = true;
        String ret = "";
        
        while(continueFlag) {
            if(strs[0].length() <= dig)
                break;
            
            char c = strs[0].charAt(dig);
            for(int k = 1; k < strs.length; k++) {
                
                if(strs[k].length() <= dig || strs[k].charAt(dig) != c) {
                    continueFlag = false;
                    break;
                }
            }
            
            if(!continueFlag)
                break;
            
            ret += c;
            dig++;
        }
        
        return ret;
    }
}

File: src/problems/014. Longest Common Prefix/Solution.js
// 14. Longest Common Prefix
// Accepted 64ms

/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if(strs.length === 0)
        return "";
    let k = 0, endFlag = false;
    
    while(!endFlag) {
        let char = strs[0].charAt(k);
        for(let str of strs) {
            if(k >= str.length || str.charAt(k) !== char) {
                endFlag = true;
                break;
            }
        }
        ++k;
    }
    
    return strs[0].slice(0, k - 1);
};
